### ISSUE TITLE
Add "exposeUnsetFields" to transformOptions

### ISSUE THREAD

**Issue by GuilhermeCunha** [nestjs/nest#6443]
## Is your feature request related to a problem? Please describe.
I would like to make sure that a certain property is not received inside my controller when it has the value "undefined".

## Describe the solution you'd like
Apparently class-tranformer has a configuration (see [here](https://github.com/typestack/class-transformer/issues/580) ) that has this feature called "exposeUnsetFields", but the "transformOptions" parameter of ValidationPipe does not have this option.

I believe that only by adding this parameter the problem would be solved.

## Teachability, Documentation, Adoption, Migration Strategy
```
app.useGlobalPipes(
        new ValidationPipe({
            transform: true,
            whitelist: true,
            transformOptions: {
                ...others,
                exposeUnsetFields: false
            }
        }),
    );
```

## What is the motivation / use case for changing the behavior?
Having this feature, it will be possible to implement several functions that will help to maintain a more correct API, being sure of what is being received, and allowing to make transformations such as: When property A is 0, completely remove (not only set it undefined) property B of the received object.

**Comment by kamilmysliwiec** [comment]
Would you like to create a PR for this feature?

**Comment by GuilhermeCunha** [comment]
I went to test this functionality, but even passing the "exposeUnsetFields" as false, the object is still arriving as "undefined" in the controller.


Would anything else need to be implemented for these settings to be passed on to the class-transformer?


![image](https://user-images.githubusercontent.com/19752833/108386204-32e42d80-71eb-11eb-835a-d51ce38a0713.png)

![image](https://user-images.githubusercontent.com/19752833/108386240-3972a500-71eb-11eb-9e1c-aa2cf0031276.png)

**Comment by Tony133** [comment]
With the pull I did the interface is aligned with that of the class-transformer package here I think it's a great addition.

NestJS use class-transformer package, I think you should ask a new question here too https://github.com/typestack/class-transformer

if you want to transform input data into the desired output you can create a custom pipe

**Comment by kamilmysliwiec** [comment]
From what I can see, it should work @GuilhermeCunha (`transformOptions` object is being passed down to the `classToPlain` function). Can you share a minimal reproduction repository?

**Comment by GuilhermeCunha** [comment]
What was causing my code not to work was the version of the class-transformer. Because of the changes that "broke" the code a few weeks ago, I was not using the most current version.

 I just installed the latest version of the class-transformer and adapted my code to accept it and everything worked as expected. 

**Thank you so much for your time!** 

I leave [here](https://github.com/GuilhermeCunha/nestjs-class-transformer) a repository with a simple example of what I needed to do working.
